mlflow/lightgbm.py

```diff
@@ -455,8 +459,9 @@ def log_feature_importance_plot(features, importance, importance_type):
             if input_example is None:
                 raise Exception("failed to gather example input.")
 
-            model_output = model.predict(input_example)
-            signature = infer_signature(input_example, model_output)
+            if log_model_signature:
+                model_output = model.predict(input_example)
+                signature = infer_signature(input_example, model_output)
         except Exception as e:  # pylint: disable=broad-except
             input_example = None
```

Comment:
For lightgbm, xgboost, and sklearn, can we stop resetting this input example field to `None` in the case where signature is unspecified? I think these concepts are independent, especially given that they can be configured independently.

Reply:
great point, now they are separate toggles so it makes no sense if the input example disappears when signature inference fails lol

Reply:
I will follow this up in another PR